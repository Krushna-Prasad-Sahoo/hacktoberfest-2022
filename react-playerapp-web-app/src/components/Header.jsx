import React from 'react'
import { NavLink } from 'react-router-dom'
export default function Header(props) {
    return (
        <nav class="navbar navbar-expand-lg navbar-dark bg-dark">
            <div>
                <>
                    <ul>
                        <li><NavLink to="/addplayer">Addplayer</NavLink></li>
                        <li><NavLink to="/showplayer">Showplayer</NavLink></li>
                    </ul>
                </>
            </div>

        </nav>
    )
}
