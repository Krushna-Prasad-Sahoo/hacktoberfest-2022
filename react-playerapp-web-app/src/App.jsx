import logo from './logo.svg';
import './App.css';
import Header from './components/Header';
import AddPlayer from './components/AddPlayer';
// import { useState, createContext, useContext } from "react";
// import { PlayerContext } from './data/PlayerContext';
import {
  BrowserRouter,
  Routes,
  Route,
 } from 'react-router-dom';
import ShowPlayer from './components/ShowPlayer';
function App() {
  // const user = useContext(PlayerContext);
  // console.log(user);
  return (
    <div className="App">
     {/* <Header/> */}
      <BrowserRouter>
      <Routes>
       <Route path="/" element={<Header/>} />
       <Route path="/addplayer" element={<AddPlayer/>} />
      <Route path="/showplayer" element={<ShowPlayer/> } />
      </Routes>
        
       </BrowserRouter>
    </div>
  );
}

export default App;
