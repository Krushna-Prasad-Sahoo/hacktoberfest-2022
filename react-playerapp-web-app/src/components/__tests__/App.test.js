 
import  App from '../../App'
import toJson from 'enzyme-to-json';
import{Route} from 'react-router-dom';
import Header from '../Header';
import Enzyme, { shallow, render, mount } from 'enzyme';
xdescribe("App testing",()=>{
  const wrapper = shallow(<App/>) 
it('renders correctly App', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('contains Route /', () => {                                       
    expect(wrapper.find(Route).at(0).props().path).toEqual('/');
   });
   it('contains Route /AddPlayer', () => {                                       
    expect(wrapper.find(Route).at(1).props().path).toEqual('/AddPlayer');
   });
   it('contains Route /ShowPlayer', () => {                                       
    expect(wrapper.find(Route).at(2).props().path).toEqual('/ShowPlayer');
   });
   it('contains Route /ShowPlayer', () => {                                       
    expect(wrapper.contains(<Header text="PlayerApp"/>)).toBeTruthy();
   });
})
