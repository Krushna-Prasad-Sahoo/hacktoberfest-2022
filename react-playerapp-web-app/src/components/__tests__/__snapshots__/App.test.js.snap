// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App testing renders correctly App 1`] = `
<div
  className="App"
>
  <BrowserRouter>
    <Header
      text="PlayerApp"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/AddPlayer"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/ShowPlayer"
    />
  </BrowserRouter>
</div>
`;
